src/Utils/PaginatorTrait.php

```diff
@@ -13,11 +18,36 @@ trait PaginatorTrait {
      * @param AbstractService $service
      * @return Paginator
      */
-    public function createPaginationForRequestWithService(RequestInterface $request, AbstractService $service): Paginator
+    public function createFromRequest(RequestInterface $request, AbstractService $service): Paginator
     {
         $page = $request->getParameter('page');
+        $totalResults = $service->countRows()["rows"];
```

Comment:
AbstractService (urasc clasa asta) nu pare sa aiba in interfata ei (nici in abstracta) metoda `countRows()`. Type hint-ul pe care il folosesti pentru un obiect ar trebui sa contina metodele pe care le vei folosi de pe obiectul respectiv.

AbstractService nu inteleg de ce exista. E o clasa goala pur si simplu. Ori o pui sa faca ceva, ori o stergi. Daca o extinzi doar de dragu de a avea un type hint comun pentru toate serviciile, atunci macar fa o interfata goala (tehnica de a face o interfata goala pentru a putea type hint-ui pe anumite obiecte se numeste Marker Interface)